src/runtime/runtime_avr.go

```diff
@@ -56,32 +61,49 @@ func postinit() {
 
 func init() {
 	initUART()
+	machine.InitMonotonicTimer()
+	nextTimerRecalibrate = ticks() + timerRecalibrateInterval
 }
 
-const tickNanos = 1024 * 16384 // roughly 16ms in nanoseconds
-
 func ticksToNanoseconds(ticks timeUnit) int64 {
-	return int64(ticks) * tickNanos
+	return int64(ticks)
 }
 
 func nanosecondsToTicks(ns int64) timeUnit {
-	return timeUnit(ns / tickNanos)
+	return timeUnit(ns)
 }
 
-// Sleep this number of ticks of 16ms.
-//
-// TODO: not very accurate. Improve accuracy by calibrating on startup and every
-// once in a while.
+// Sleep this number of ticks of nanoseconds.
 func sleepTicks(d timeUnit) {
-	currentTime += d
-	for d != 0 {
-		sleepWDT(WDT_PERIOD_16MS)
-		d -= 1
+	waitTill := ticks() + d
+	// recalibrate if we have some time (>100ms) and it was a while when we did it last time.
+	if d > 100000 {
+		now := waitTill - d
+		if nextTimerRecalibrate < now {
+			nextTimerRecalibrate = now + timerRecalibrateInterval
+			machine.AdjustMonotonicTimer()
+		}
+	}
```

Comment:
@aykevl, you had added TODO comment some time back, and I have added code to recalibrate timer with interval 1 minute when sleep is called for 100ms or more. Do you think one minute interval is too short?

Reply:
I doubt we need to recalibrate more often. Pretty sure one minute is sufficient.

Reply:
But `AdjustMonotonicTimer` doesn't calibrate anything? Therefore calling it every minute is totally unnecessary.

In my comment I was referring to calibrating the WDT based on the timer. The timer is already accurate (or as accurate as it can get).